crates/core/src/operations/unset_tbl_properties.rs

```diff
@@ -0,0 +1,169 @@
+//! Unset table properties of the table
+
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use chrono::Utc;
+use futures::future::BoxFuture;
+use serde_json::json;
+
+use crate::kernel::{Action, CommitInfo, IsolationLevel, Protocol};
+use crate::logstore::LogStoreRef;
+use crate::operations::transaction::commit;
+use crate::protocol::DeltaOperation;
+use crate::table::config::DeltaConfigKey;
+use crate::table::state::DeltaTableState;
+use crate::DeltaTable;
+use crate::{DeltaResult, DeltaTableError};
+
+/// Remove constraints from the table
+pub struct UnsetTablePropertiesBuilder {
+    /// A snapshot of the table's state
+    snapshot: DeltaTableState,
+    /// Name of the property
+    properties: Option<Vec<String>>,
```

Comment:
we could just make this a `Vec<DeltaConfigKey>` and omit the `Option` checking and key validation logic below.

Reply:
That's also fine, I will still have validate the keys than in the Python binding